src/shogun-core-main/src/main/java/de/terrestris/shoguncore/service/GeoServerInterceptorService.java

```diff
@@ -353,34 +378,51 @@ public Response interceptWmtsRequest(HttpServletRequest request, String serviceI
      * @throws IOException
      */
     public Response interceptGeoServerRequest(HttpServletRequest request)
-        throws InterceptorException, URISyntaxException,
-        HttpException, IOException {
-        return interceptGeoServerRequest(request, Optional.empty());
+            throws InterceptorException, URISyntaxException, HttpException, IOException {
+        return interceptGeoServerRequest(request, new HashMap<String, Optional<String>>());
     }
 
     /**
+     *
      * @param request
-     * @param endpoint
+     * @param optionals
      * @return
      * @throws InterceptorException
      * @throws URISyntaxException
      * @throws HttpException
      * @throws IOException
      */
-    public Response interceptGeoServerRequest(HttpServletRequest request, Optional<String> endpoint)
+    public Response interceptGeoServerRequest(
+        HttpServletRequest request,
+        HashMap<String, Optional<String>> optionals)
```

Comment:
Can we have the `optionals` typed?

Reply:
not sure here, do you want me to instantiate the hashmap with null values as class variable?

Reply:
I was thinking of a simple class definition for `optionals` that defines the structure of the parameter.